src/server/api/routers/userTrips.ts

```diff
@@ -0,0 +1,69 @@
+import { z } from "zod";
+
+import { createTRPCRouter, protectedProcedure } from "../trpc";
+
+export const userTripsRouter = createTRPCRouter({
+  /**
+   * query to get all trips
+   * only query trips which the user has created or participated in
+   */
+  getAll: protectedProcedure.query(async ({ ctx }) => {
+    try {
+      return await ctx.prisma.trip.findMany({
+        select: {
+          id: true,
+          name: true,
+          description: true,
+        },
+        where: {
+          participants: {
+            some: {
+              id: ctx.session.user.id,
+            },
+          },
+        },
+      });
+    } catch (error) {
+      console.error("error", error);
+    }
+  }),
+
+  /**
+   * mutation to create a new trip
+   * @param name - name of the trip
+   * @param description - description of the trip
+   * @param startDate - start date of the trip
+   * @param endDate - end date of the trip
+   * @param adminId - id of the user who created the trip
+   * @param participants - id of users who are participating in the trip
+   */
+  createTrip: protectedProcedure
+    .input(
+      z.object({
+        name: z.string().min(3).max(50),
+        description: z.string().min(3).max(100),
```

Comment:
Description length should be higher upto 1000
```suggestion
        description: z.string().min(3).max(1000),
```